File: onyx-remote-driver/src/main/java/com/onyx/client/exception/ServerClosedException.java
package com.onyx.client.exception;

/**
 * Created by tosborn1 on 6/25/16.
 *
 * This exception indicates the server has closed
 */
public class ServerClosedException extends OnyxServerException
{
    private static final String SERVER_CLOSED_MESSAGE = "Client connection closed prematurely.";

    /**
     * Default Constructor
     * @param cause The root cause as to why the connection has been closed
     */
    public ServerClosedException(Throwable cause)
    {
        super(SERVER_CLOSED_MESSAGE, cause);
    }

}

File: onyx-remote-driver/src/main/kotlin/com/onyx/exception/ServerClosedException.kt
package com.onyx.exception

/**
 * Created by Tim Osborn on 6/25/16.
 *
 * This exception indicates the server has closed
 */
class ServerClosedException(cause: Throwable) : OnyxServerException(SERVER_CLOSED_MESSAGE, cause) {
    companion object {
        private val SERVER_CLOSED_MESSAGE = "Client connection closed prematurely."
    }
}
